File: Travel Package management C++/Destination.cpp
#include "Destination.h"
#include "Activity.h"
#include <iostream>

using namespace std;  



Destination::Destination(string n){
    name = n;
}


bool Destination::addActivity(shared_ptr<Activity> activity) {
    if (activity->isAssignedToDestination()) {
        return false;
    }
    activity->assignToDestination(name); 
    activities.push_back(activity); 
    return true;
}

void Destination::printDetails(){
    cout << "Destination name: " << name << endl;
    cout << "Activities: " << endl;
    for(const auto& activity : activities){
        activity->printDetails();
    }
}

void Destination::printAvailableActivities(){
    cout << "Destination: " << name << endl;
    cout << "Available activities: " << endl;
    for(const auto& activity : activities){
        if(activity->isAvailable()){
            activity->printAvailableActivityDetails();
        }
    }
}

string Destination::getName(){ return name; }

File: Travel Package Management Java/src/com/example/travel/Destination.java
package com.example.travel;

import java.util.ArrayList;
import java.util.List;

public class Destination {
    private String name;
    private List<Activity> activities = new ArrayList<>();

    public Destination(String name) {
        this.name = name;
    }

    public boolean addActivity(Activity activity) {
        if (activity.isAssignedToDestination()) {
            return false;
        }
        activity.assignToDestination(name);
        activities.add(activity);
        return true;
    }

    public void printDetails() {
        System.out.println("Destination name: " + name);
        System.out.println("Activities: ");
        for (Activity activity : activities) {
            activity.printDetails();
        }
    }

    public void printAvailableActivities() {
        System.out.println("Destination: " + name);
        System.out.println("Available activities: ");
        for (Activity activity : activities) {
            if (activity.isAvailable()) {
                activity.printAvailableActivityDetails();
            }
        }
    }

    public String getName() {
        return name;
    }
}
